### ISSUE TITLE
chapter_7:skip_gram and cbow programmes are wrong

### ISSUE THREAD

**Issue by Alpha-ma** [nfmcclure/tensorflow_cookbook#69]
when reading codes for this algorithm, I found one of the most important step: calculate similarity( cosine) (in both skip gram and cbow) is wrong, as it is just generated randomly, and never updated when training, you can find the results printed are not different at any time.  i.e., the results is wrong. please kindly correct this crucial error, thanks. @nfmcclure @jimthompson5802 @slcott @havk64 @cxxgtxy

**Comment by nfmcclure** [comment]
Hi @Alpha-ma ,  Thanks for bringing this to my attention.  I suspect the algorithm is performing correctly, but the hyperparameters and data pipeline are not tuned correctly.  I'll check this out soon.

**Comment by nfmcclure** [comment]
I've been meaning to address this. I didn't know a month could be this busy. This is next on my list. Hopefully I'll get some time in the next few weeks to improve the convergence of the algorithm.

**Comment by Jhonsonzhangxing** [comment]
@Alpha-ma @nfmcclure , Hi,guys, I thought  i  found where the mistake occur,in the function text_to_numbers(sentences, word_dict): for this line:for word in sentence may be wrong ,I thought the line should be for word in sentence.split().  yes ,you need add ".split()" .only in this way ,text_data=[[551,
  2588,
  2359,
  0,
  33,
  7418,
  214,
  149,......]]

for word in sentence:
     you got word=['r','o','c','k'..........],not [rock, ....]

if i was wrong please @ me ,thank you

**Comment by nfmcclure** [comment]
@Jhonsonzhangxing ,  thanks for noticing this!!  The embeddings were not converging because they were running on characters, not words.  I fixed this and will push the changes along with other changes today.  I'll close this in a few hours.

**Comment by nfmcclure** [comment]
Everything seems to be in working order.  I did not do a full test, but if you run the skipgram (learning rate between 0.5 and 1.0) for about 300,000-500,000 iterations you should get some convergence. I ran it for 100,000 and am seeing the "nearby" words change. Remember that this is a problem specific embedding (movie reviews) and that we won't end up with nice word similarities like "love" and "affection" and "like" being close to each other all the time.
